BUG: pypy39 wheel builder failing

The wheel builder workflow has been failing for pypy3.9 since last week, [see here](https://github.com/numpy/numpy/actions/workflows/wheels.yml?query=branch%3Amain).

The errors look like:

```
________________ TestNewBufferProtocol.test_error_too_many_dims ________________
  [gw1] linux -- Python 3.9.17 /tmp/tmp.0jQGXevz9m/venv/bin/python
  
  self = <numpy._core.tests.test_multiarray.TestNewBufferProtocol object at 0x0000000006a06368>
  
      @pytest.mark.slow
      def test_error_too_many_dims(self):
          def make_ctype(shape, scalar_type):
              t = scalar_type
              for dim in shape[::-1]:
                  t = dim * t
              return t
      
          # Try constructing a memory with too many dimensions:
          c_u8_65d = make_ctype((1,)*65, ctypes.c_uint8)
          try:
              m = memoryview(c_u8_65d())
          except ValueError:
              pytest.skip("memoryview doesn't support more dimensions")
      
          assert_equal(m.ndim, 65)
      
          with pytest.raises(RuntimeError, match=".*ndim"):
  >           np.array(m)
  E           Failed: DID NOT RAISE <class 'RuntimeError'>
```

And I think that was due to #25149.

--- a/numpy/_core/tests/test_multiarray.py
+++ b/numpy/_core/tests/test_multiarray.py
@@ -8205,33 +8205,6 @@ class TestNewBufferProtocol:
         a = np.ones((1,) * 32)
         self._check_roundtrip(a)
 
-    @pytest.mark.slow
-    def test_error_too_many_dims(self):
-        def make_ctype(shape, scalar_type):
-            t = scalar_type
-            for dim in shape[::-1]:
-                t = dim * t
-            return t
-
-        # Try constructing a memory with too many dimensions:
-        c_u8_65d = make_ctype((1,)*65, ctypes.c_uint8)
-        try:
-            m = memoryview(c_u8_65d())
-        except ValueError:
-            pytest.skip("memoryview doesn't support more dimensions")
-
-        assert_equal(m.ndim, 65)
-
-        with pytest.raises(RuntimeError, match=".*ndim"):
-            np.array(m)
-
-        # The above seems to create some deep cycles, clean them up for
-        # easier reference count debugging:
-        del c_u8_65d, m
-        for i in range(65):
-            if gc.collect() == 0:
-                break
-
     def test_error_pointer_type(self):
         # gh-6741
         m = memoryview(ctypes.pointer(ctypes.c_uint8()))

